packages/universal-router-sdk/src/entities/protocols/pancakeswap.ts

```diff
@@ -299,89 +299,76 @@ async function addMixedSwap(
     } else {
       const sections = SmartRouter.partitionMixedRouteByProtocol(route)
 
-      const isLastSectionInRoute = (i: number) => {
-        return i === sections.length - 1
-      }
-
       let outputToken
       let inputToken = inputAmount.currency.wrapped
 
       for (let i = 0; i < sections.length; i++) {
         const section = sections[i]
+        const nextSection = sections[i + 1] ?? []
+        const isFirstSection = i == 0
+        const isLastSection = i === sections.length - 1
+
+        const nextIsV2 = nextSection.every(SmartRouter.isV2Pool)
```

Comment:
`nextIsV2` will be true if `nextSection` is an empty array?

Reply:
nope, 
```ts
export function isV2Pool(pool: Pool): pool is V2Pool {
  return pool.type === PoolType.V2
}
```

Reply:
em... wait
<img width="190" alt="image" src="https://github.com/pancakeswap/pancake-frontend/assets/144641937/c22cc7a6-e6a8-4bfd-957a-8c4a8b634284">